### PR TITLE
Validations model specs

### PR DESCRIPTION
**Pull request by NoelLincoln**
Hello 👋,

In this milestone I was able to use validations for models and write unit tests for models.

I achieved the following as required:

**Add the following validations:**
**For the User model:**

- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.

**For the Post model:**

- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.

Add unit specs for all of your models' methods(comment and like) and validations.

Cheers!!

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -1,4 +1,13 @@
...
   belongs_to :user
   belongs_to :post
+
+  after_create :update_post_comments_counter
+
+  private
+
+  def update_post_comments_counter
+    post&.update(comments_counter: post.comments.count)
+  end
```
Comment (Efiamotu-1) [inline app/models/comment.rb:12]: - [ ] To meet the below requirement please add unit tests for this custom method created for the comment model :+1: 
mm

File: app/models/like.rb [hunk 2]
```diff
@@ -1,4 +1,12 @@
...
   belongs_to :user
   belongs_to :post
+
+  after_create :update_post_likes_counter
+
+  private
+
+  def update_post_likes_counter
+    post&.update(likes_counter: post.likes.count)
+  end
```
Comment (Efiamotu-1) [inline app/models/like.rb:11]: - [ ] To meet the below requirement please add unit tests for this custom method created for the like model :+1: 
mm

File: app/models/post.rb [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
+  after_create :update_user_posts_counter
+
+  def update_user_posts_counter
+    author&.update(posts_counter: author.posts.count)
+  end
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
```
Comment (Efiamotu-1) [inline app/models/post.rb:17]: - [ ] To meet the below requirement please add unit tests for this custom method created for the post model :+1: 
mm

File: app/models/user.rb [hunk 4]
```diff
@@ -1,5 +1,13 @@
...
-  has_many :posts
+  has_many :posts, foreign_key: 'author_id'
   has_many :likes
+
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
```
Comment (Efiamotu-1) [inline app/models/user.rb:12]: - [ ] To meet the below requirement please add unit tests for this custom method created for the user model :+1: 
mm

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @NoelLincoln , 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/comment.rb (hunk 1 above)
Comment (NoelLincoln) [reply app/models/comment.rb:12]: Thanks, I must have forgotten that bit.

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @NoelLincoln 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- You added tests for all custom methods :heavy_check_mark: 
- All tests are passing :heavy_check_mark:  

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**